### PR TITLE
fix: subgraph queries responses

### PR DESCRIPTION
**Pull request by aasseman**
Mostly to make it behave like the original `indexer-service`.

### CODE REVIEW

File: service/src/query_processor.rs [hunk 1]
```diff
@@ -71,20 +71,25 @@ impl ToString for SubgraphDeploymentID {
...
+    pub graphQLResponse: String,
+    pub attestation: Option<Signature>,
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[allow(non_snake_case)] // Need exact field names for JSON response
 pub struct UnattestedQueryResult {
-    pub graphql_response: String,
+    pub graphQLResponse: String,
```
Comment (hopeyen) [inline service/src/query_processor.rs:92]: Perhaps we can use serde alias here to match JSON response?

File: service/src/query_processor.rs [hunk 2]
```diff
@@ -100,6 +105,7 @@ pub struct Response<T> {
 #[derive(Debug)]
 pub struct FreeQuery {
     pub subgraph_deployment_id: SubgraphDeploymentID,
+    // TODO: Use Arc<String> instead of String?
```
Comment (hopeyen) [inline service/src/query_processor.rs:108]: maybe worthwhile, especially if you think the query string is shared across threads or parts of the program, or if it is used multiple times, or should be long living

File: service/src/query_processor.rs (hunk 1 above)
Comment (aasseman) [reply service/src/query_processor.rs:92]: Oh cool didn't know about that solution. I know `strum` can do that too, I used it in the TAP Aggregator service.

File: service/src/query_processor.rs (hunk 2 above)
Comment (aasseman) [reply service/src/query_processor.rs:108]: Yup, in particular I noticed that
https://github.com/hopeyen/indexer-service-rs/blob/c604c7a258fad37ef74abd20823571d5f4db9e3e/service/src/graph_node.rs#L28-L32
needs the cloning.

**Comment by aasseman** [comment]
Rebased onto latest main

File: service/src/query_processor.rs (hunk 1 above)
Comment (aasseman) [reply service/src/query_processor.rs]: Fix in [0c02b88](https://github.com/graphops/indexer-service-rs/pull/21/commits/0c02b88971474da1147ffd1ca8565fefbe2c9378)

**Review by hopeyen — APPROVED** [review]
🐈‍⬛  (might need another rebase to main:p

File: service/src/query_processor.rs (hunk 2 above)
Comment (aasseman) [reply service/src/query_processor.rs]: I played around, went down the rabbit hole, and gave up: [8b9b531](https://github.com/graphops/indexer-service-rs/pull/21/commits/8b9b531ea6b59ef34563dc18ca4f9590727890a1)

Comment (hopeyen) [reply service/src/query_processor.rs]: haha we could leave the TODO comment in for a future refactor potentially?

Comment (aasseman) [reply service/src/query_processor.rs]: Well, I'm not even sure that Arc is the solution here. So I'm of the opinion of sticking to the YAGNI rule, we can do something about it when we'll actually need it.

**Review by hopeyen — APPROVED** [review]
🚢

**Pull request merged**